src/plugins/vis_builder/public/application/utils/state_management/visualization_slice.ts

```diff
@@ -123,7 +123,7 @@ export const slice = createSlice({
     builder.addCase(setActiveVisualization, (state, action) => {
       state.activeVisualization = {
         name: action.payload.name,
-        aggConfigParams: [],
+        aggConfigParams: action.payload.aggParams,
```

Comment:
Would this benefit from `|| []`?

Reply:
I think it is not necessary to add || [] here? since if there is no aggregation configs being added when we switch viz type, the aggConfigParams will already be an empty array.